src/utils/checkreq.ts

```diff
@@ -0,0 +1,86 @@
+import { Types } from "mongoose";
+interface IParameter {
+  field: string;
+  type: string | null | undefined | unknown | Types.ObjectId;
+}
+const checkNotAllowedParameters = (
+  object: Record<
+    string,
+    object | string | number | boolean | null | undefined | unknown
+  >,
+  allowedParameters: string[],
+  allowedTypes: (string | number | boolean | null | undefined | unknown)[]
+) => {
+  const incorrectProperties = allowedParameters.filter((key, index) => {
+    const type = allowedTypes[index];
+    const value = object[key];
+    if (type === Types.ObjectId) {
+      if (!Types.ObjectId.isValid(value as string)) {
+        throw new Error("el id no es un Object Id");
+      }
+      return false;
+    }
+
+    return typeof value !== type;
+  });
+
+  if (incorrectProperties.length > 0) {
+    throw new Error(
+      "Properties with incorrect types:" + incorrectProperties.join(",")
+    );
+  }
+  return;
+};
+const checkRequiredParameters = (
+  object: Record<
+    string,
+    object | string | number | boolean | null | undefined | unknown
+  >,
+  allowedParameters: string[]
+) => {
+  const missingParameters = allowedParameters.filter(
+    (key) => !(key in object) || object[key] == null || object[key] === ""
```

Comment:
Evitar en lo posible las igualaciones con coerción -> ==

Qué pasaría en el caso en que yo quiera que el objeto tenga una propiedad con `null` o que fuera un string vacío `""`?

Ambos son valores válidos, lo que deberían chequear para que sea requerido es simplemente que el valor sea diferente a `undefined`

```suggestion
    (key) => !(key in object) || object[key] === undefined
```